codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -160,6 +160,23 @@ private static boolean validateHeadersSentState(Http2Stream stream, Http2Headers
         return isInformational;
     }
 
+    /**
+     * According to RFC 9113, "Pseudo-header fields defined for requests MUST NOT appear in responses;
+     * pseudo-header fields defined for responses MUST NOT appear in requests."
+     * @return {@code true} if validation completed successfully, {@code false} if not
+     */
+    private boolean validatePseudoHeaders(Http2Headers headers) {
+        boolean isResponseHeaders = connection.isServer();
+        for (CharSequence name : headers.names()) {
```

Comment:
headers.names() copies names into new `LinkedHashSet` - for iteration It is sufficient to use Http2Headers.iterator() which does not do redundant copy (possibly guarded by `isEmpty()` check returning `Collections.emptyIterator()`)

Reply:
Will make the change, thanks for the comment!